File: Syeew/Utils/DTOs/RequestDataDTO.cs
namespace Syeew.Utils.DTOs
{
    public class RequestDataDTO
    {
        public string CompanyName { get; set; }
        public CustomDate From { get; set; }

        public CustomDate To { get; set; }

        public string ContentX { get; set; }

        public string ContentY { get; set; }
    }
}

File: Syeew/ClientApp/src/app/Utils/DTOs/RequestDataDTO.ts
import { CustomDate } from "../CustomDate";

export class RequestDataDTO {
    companyName!: string;
    from!: CustomDate;
    to!: CustomDate;
    contentY!: string;
    contentX?: string;

    constructor(companyName: string, dateFrom: CustomDate, dateTo: CustomDate, contentY: string, contentX?: string) {
        this.companyName = companyName
        this.from = dateFrom
        this.to = dateTo
        this.contentY = contentY
        if (contentX != undefined)
            this.contentX = contentX
        else
            this.contentX = ''
    }

}